compiler/src/dotty/tools/dotc/ast/Positioned.scala

```diff
@@ -237,8 +240,14 @@ abstract class Positioned(implicit @constructorOnly src: SourceFile) extends Src
 }
 
 object Positioned {
-  @sharable private[Positioned] var debugId = Int.MinValue
+  @sharable private var debugId = Int.MinValue
+  @sharable private var ids: java.util.WeakHashMap[Positioned, Int] = null
+  @sharable private var nextId: Int = 0
 
-  def updateDebugPos(using Context): Unit =
+  def init(using Context): Unit =
     debugId = ctx.settings.YdebugTreeWithId.value
+    if ctx.settings.YshowTreeIds.value
```

Comment:
```suggestion
    if ids == null && ctx.settings.YshowTreeIds.value
```

Reply:
No, I think we want to reset ids on each `init`. I am going to add an `else` clause that sets `ids = null`.

Reply:
Or, do you think it is better to keep ids once it is set? That would allow us to track ids across multiple runs, which might be useful.